src/Psr/CacheItemPool/CacheItemPoolDecorator.php

```diff
@@ -84,7 +83,7 @@ public function __destruct()
     /**
      * {@inheritdoc}
      */
-    public function getItem($key)
+    public function getItem($key): CacheItemInterface
```

Comment:
Can we also add argument types so we are aligned with the interface?

Reply:
added (needed to drop "psr/cache": "^1.0  for it to work)

Reply:
Adding the return type here is still a BC break :| 

We're still in a "new major target" scenario, I fear...

Reply:
Most likely previously it was just return type in documentation but support for psr/cache v2 and v3 it needs to be added.

Reply:
Yup: not blaming your changes, just saying that the lack of `final` has exposed us to BC issues here, since the dependency change leads to a transitive BC break.

Reply:
I see no changes to support all three versions of PSR-6. Therefore it would be good to have plan before things are constantly being changed here.
@boesing, you as maintainer of this component, what is your plan for a new major version?

Reply:
Oh, I've expected the decorators to be final since v3.
Seems that I forgot these...

I would be open for having a new major but only if we also add support for `psr/simple-cache` in its newer versions.

In the next major we can add `final` to the decorators as well. So we do have to drop v1 support of the psr packages as well?

Reply:
> I would be open for having a new major but only if we also add support for `psr/simple-cache` in its newer versions.

👍🏻

Reply:
So what is the plan with this one? Does it mean that v4 will happen in September?

Reply:
I do not give an ETA but I will take care that this lands in v4.

Since that requires all supported adapters to support v4 as well plus the fact that there are some other things I've planned for v4, I dont think this will happen in september.

Reply:
Thank you for your response let me know how I can help with v4.

If you will create v4 branch  I will create new pr against it and will fix this code for it.